rustls-mbedtls-provider-utils/src/pk.rs

```diff
@@ -75,6 +75,21 @@ pub fn rustls_signature_scheme_to_mbedtls_curve_id(signature_scheme: SignatureSc
     }
 }
 
+/// Helper function to get [`rustls::SignatureAlgorithm`] from mbedtls [`mbedtls::pk::Type`]
+pub fn pk_type_to_signature_algo(pk_type: Type) -> rustls::SignatureAlgorithm {
+    use rustls::SignatureAlgorithm;
+    match pk_type {
+        Type::Rsa => SignatureAlgorithm::RSA,
+        Type::Ecdsa => SignatureAlgorithm::ECDSA,
+        Type::RsassaPss => SignatureAlgorithm::RSA,
+        Type::RsaAlt => SignatureAlgorithm::RSA,
+        Type::Eckey => SignatureAlgorithm::ECDSA,
+        Type::EckeyDh => SignatureAlgorithm::Unknown(u8::MAX),
```

Comment:
Can we have this function return `Option<SignatureAlgorithm>` instead?

Reply:
sure

Reply:
done